api.go

```diff
@@ -119,7 +120,8 @@ func (api *API) Query(ctx context.Context, req *QueryRequest) (QueryResponse, er
 	resp.Results = results
 
 	// Fill column attributes if requested.
-	if req.ColumnAttrs && !req.ExcludeColumns {
+	// execOpts.ColumnAttrs is an out param from api.server.executor.Execute
```

Comment:
I'm a little confused by how this comment is worded. Are you trying to say that because execOpts is passed to Execute as a pointer, it may modify execOpts.ColumnAttrs and that modified value would be used here?

Reply:
Right, `execOpts.ColumnAttrs` is returned from `Execute`.

Reply:
saying it's "returned" is confusing IMO since it isn't actually a return value of that method. The method modifies execOpts as a side effect of its execution which really isn't the friendliest thing to reason about. Can we change the comment to "// execOpts.ColumnAttrs may be set by the Execute method if any of the Calls use Options(columnAttrs=true)"?

Hopefully once https://github.com/pilosa/pilosa/issues/1632 is implemented we won't need this behavior or that comment at all.

Reply:
Well, returning a value is what it does. It's like C, where pointer arguments are used to return more than one value.

Reply:
I see where you're coming from, but I strongly disagree with use of "return" in this context. It explicitly does *not* return this value in C or Go though I agree that "filling" a struct in this way is a common pattern in C and even Go in some cases. In this case we're not even passing a pointer to an empty struct and then filling it,  we're passing in options which the method both reads and writes to so that they may be used by code after its execution. This was confusing to me and I think more precise terminology would have helped me understand more easily.

Reply:
ok, updated.